### PR TITLE
feat(cli): Add tail flag to the log command

### LINKED ISSUE #3004: kamel log tail option
We're currently developing multiple integrations in dev environment so we need to watch logs very often, but the amount of logs retrieve from the container everytime we issue the command has started to annoy some of us.

Will be nice to have a `--tail` option just like kubectl/oc commands.

### PR DESCRIPTION
**Pull request by essobedo**
fixes #3004

## Motivation

When a pod has a lot of log entries, it can be convenient to be able to get only the last entries of the logs.

## Modifications:

* Add an int flag `--tail` to get the specified amount of last lines of the logs. The default value is `-1` indicating that all lines of the logs are expected like the previous default behavior

**Release Note**
```release-note
feat(cli): Add tail flag to the log command
```

### CODE REVIEW

**Review by squakez — COMMENTED** [review]
Nice one! Would be the default behavior the same as it was before introducing the feature?

**Comment by essobedo** [comment]
@squakez yes, the default behavior remains the same

**Comment by essobedo** [comment]
The 3 violations raised are not related to these changes https://github.com/apache/camel-k/pull/3589/files#annotation_4463020346

**Comment by essobedo** [comment]
The only failure is on `TestNativeIntegrations/automatic_rollout_deployment_from_fast-jar_to_native_kit` which is a known failing test

**Review by squakez — APPROVED** [review]
Really nice feature, thanks!

File: pkg/cmd/log.go
Comment (tadayosi) [inline pkg/cmd/log.go:142]: When `o.TailLines` is explicitly defaulted to `10`, do we really need this `if`?  Also, if we do, you should use `pointer.Into64(10)` here:
```
tailLines = &[]int64{10}[0]
```

**Review by tadayosi — CHANGES_REQUESTED** [review]
A small change request.

File: pkg/cmd/log.go
Comment (essobedo) [reply pkg/cmd/log.go:140]: Good catch, thx. Fixed

**Review by tadayosi — APPROVED** [review]

**Comment by essobedo** [comment]
BTW how do you proceed in camel-k in terms of PR validation? I mean: How many PR approvals are needed? Who merges it?

File: pkg/cmd/log.go
Comment (tadayosi) [inline pkg/cmd/log.go:50]: I noticed that `-f` shorthand generally means "follow" as in `kubectl logs -f` and our `kamel log` has this behaviour. So `-f` may not be a good shorthand character. `-t` is also not a good shorthand as it's already used in `kamel run` as `--trait`. Maybe we should simply force `--tail`.

**Review by tadayosi — CHANGES_REQUESTED** [review]

File: pkg/cmd/log.go
Comment (essobedo) [reply pkg/cmd/log.go:50]: I agree, let's do that

Comment (tadayosi) [reply pkg/cmd/log.go:50]: Or like `kubectl logs` we could possibly drop the original `--tail` and always force passing line numbers with `--tail` without a default value:
https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/logs/logs.go

Comment (essobedo) [reply pkg/cmd/log.go:50]: Good idea, let's use the same behavior as `kubectl logs`:
```
    --tail=-1:
	Lines of recent log file to display. Defaults to -1 with no selector, showing all log lines otherwise 10, if a
	selector is provided.
```

**Comment by tadayosi** [comment]
I believe we are on hold for merging pull reqs which add a feature and are not related to stabilising e2e just because we've been in the process of releasing 1.10.0 and moving to 1.11.0-SNAPSHOT. When everything is set up, we could resume merging them.

**Comment by essobedo** [comment]
> I believe we are on hold for merging pull reqs which add a feature and are not related to stabilising e2e just because we've been in the process of releasing 1.10.0 and moving to 1.11.0-SNAPSHOT. When everything is set up, we could resume merging them.

I see it is a specific situation, but what about the process in a normal situation?

**Review by tadayosi — APPROVED** [review]
Great. Thanks.

**Comment by tadayosi** [comment]
> I see it is a specific situation, but what about the process in a normal situation?

@essobedo Basically, once a PR has enough approvals and all CI checks have passed anyone with "Member" right to this repo could merge it. Practically, @squakez, I, and @oscerd would take care of them for most cases these days.

**Comment by oscerd** [comment]
By the way since @essobedo is a committer, he could do it by himself.

**Comment by essobedo** [comment]
> By the way since @essobedo is a committer, he could do it by himself.

Yes I know but I'm still new to the project and I'm not a Go expert that's why I'm asking

**Comment by squakez** [comment]
> > By the way since @essobedo is a committer, he could do it by himself.
> 
> Yes I know but I'm still new to the project and I'm not a Go expert that's why I'm asking

No worries. If the PR was approved by some committer and checks passed, then, you can merge yourself. I usually wait 24/48h more to see any other review. Also, I prefer to apply "rebase and merge" strategy in order to have a linear story of the commits on the main branch.

**Comment by tadayosi** [comment]
Looks all good. Let's merge.

**Pull request merged**
